lib/puppet-lint/lexer.rb

```diff
@@ -123,7 +123,7 @@ def heredoc_queue
     [:WHITESPACE, %r{\A(#{WHITESPACE_RE}+)}],
     # FIXME: Future breaking change, the following :TYPE tokens conflict with
     #        the :TYPE keyword token.
-    [:TYPE, %r{\A(Integer|Float|Boolean|Regexp|String|Array|Hash|Resource|Class|Collection|Scalar|Numeric|CatalogEntry|Data|Tuple|Struct|Optional|NotUndef|Variant|Enum|Pattern|Any|Callable|Type|Runtime|Undef|Default|Sensitive)\b}], # rubocop:disable Layout/LineLength
+    [:TYPE, %r{\A(Any|Array|Boolean|Callable|CatalogEntry|Class|Collection|Data|Default|Enum|Error|Float|Hash|Integer|NotUndef|Numeric|Optional|Pattern|Regexp|Resource|Runtime|Scalar|Sensitive|String|Struct|Tuple|Type|Undef|Variant)\b}], # rubocop:disable Layout/LineLength
```

Comment:
I like the sorting by name, but could you please do that in a second commit? That makes it a bit easier to read the diff.

Reply:
Will do in a few days

Reply:
I split the change in to 2 commits. Please enjoy :-D